akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/AccountExample20.java

```diff
@@ -0,0 +1,150 @@
+      this.amount = amount;
+    }
+  }
+  public static class CloseAccount implements AccountCommand {}
+
+  interface AccountEvent {}
+  public static class AccountCreated implements AccountEvent {}
+  public static class Deposited implements AccountEvent {
+    public final double amount;
+
+    Deposited(double amount) {
+      this.amount = amount;
+    }
+  }
+  public static class Withdrawn implements AccountEvent {
+    public final double amount;
+
+    Withdrawn(double amount) {
+      this.amount = amount;
+    }
+  }
+  public static class AccountClosed implements AccountEvent {}
+
+  interface Account {}
+  public static class EmptyAccount implements Account {
+    @CmdHandler
+    public Effect<AccountEvent, Account> create(CreateAccount cmd) {
+      return Effects.persist(new AccountCreated());
+    }
+
+    @EvtHandler
+    public Account created(AccountCreated evt) {
+      return new OpenedAccount(0.0);
+    }
+  }
+  public static class OpenedAccount implements Account {
+    public final double balance;
+
+    OpenedAccount(double balance) {
+      this.balance = balance;
+    }
+
+    @CmdHandler
+    public Effect<AccountEvent, Account> deposit(Deposit cmd) {
+      return Effects.persist(new Deposited(cmd.amount));
+    }
+
+    @CmdHandler
+    public Effect<AccountEvent, Account> withdraw(Withdraw cmd) {
+      if ((balance - cmd.amount) < 0.0) {
+        return Effects.unhandled(); // TODO replies are missing in this example
+      } else {
+        // Type inference doesn't work well here, Effects.<AccountEvent, Account> persist(
+        return Effects.<AccountEvent, Account> persist(new Withdrawn(cmd.amount))
+            .andThen(acc2 -> {
+              // we know this cast is safe, but somewhat ugly
+              OpenedAccount openAccount = (OpenedAccount) acc2;
+              // do some side-effect using balance
+              System.out.println(openAccount.balance);
+            });
+      }
+    }
+
+    @CmdHandler
+    public Effect<AccountEvent, Account> close(CloseAccount cmd) {
+      if (balance == 0.0)
+        return Effects.persist(new AccountClosed());
+      else
+        return Effects.unhandled();
+    }
+
+    @EvtHandler
+    public Account deposited(Deposited evt) {
+      return new OpenedAccount(balance + evt.amount);
+    }
+
+    @EvtHandler
+    public Account withdrawn(Withdrawn evt) {
+      return new OpenedAccount(balance - evt.amount);
+    }
+
+    @EvtHandler
+    public Account closed(AccountClosed evt) {
+      return new ClosedAccount();
+    }
+  }
+  public static class ClosedAccount implements Account {
+
+    @CmdHandler
+    public Effect<AccountEvent, Account> anyCommand(AccountCommand cmd) {
+      // not necessary to have this handler, default is also unhandled, unless we want to warn about missing handlers
+      return Effects.unhandled();
+    }
+
+  }
+
+  public static class AccountEntity extends PersistentBehavior<AccountCommand, AccountEvent, Account> {
```

Comment:
Huh...This isn't quite how I would have envisioned it. I would have put the @CmdHandler and @EvtHandler annotations on methods inside of the `AccountEntity`, rather than inside of the `Account`.

However, the approach of putting it inside of the `Account` is interesting as well. Would we be able to do either?

Reply:
I was thinking that we would support both, see the other alternative in AccountExample21